src/hooks/useHtmlPaste/index.ts

```diff
@@ -16,8 +16,12 @@ const insertAtCaret = (target: HTMLElement, text: string) => {
         range.insertNode(node);
 
         // Move caret to the end of the newly inserted text node.
-        range.setStart(node, node.length);
-        range.setEnd(node, node.length);
+        // we have to set offset as `node.length - 1` when text has empty line at the end
```

Comment:
```suggestion
        // Correctly calculate text length since newline character does not create a new line in HTML
        // https://github.com/Expensify/App/issues/42216
```